dspace-api/src/main/java/org/dspace/statistics/AnonymizeStatistics.java

```diff
@@ -0,0 +1,274 @@
+
+public class AnonymizeStatistics {
+
+    private static Logger log = getLogger(AnonymizeStatistics.class);
+    private static Context context = new Context();
+    private static String action = "anonymise_statistics";
+
+    private static final String HELP_OPTION = "h";
+    private static final String SLEEP_OPTION = "s";
+    private static final String BATCH_SIZE_OPTION = "b";
+    private static final String THREADS_OPTION = "t";
+
+    private static int sleep;
+
+    private static SolrLoggerService solrLoggerService =
+            StatisticsServiceFactory.getInstance().getSolrLoggerService();
+    private static ConfigurationService configurationService =
+            DSpaceServicesFactory.getInstance().getConfigurationService();
+
+    private static int batchSize = 100;
+    private static int threads = 2;
+
+    private static final Object ANONYMISED =
+            configurationService.getProperty("anonymise_statistics.dns_mask", "anonymised");
+
+    private static final String TIME_LIMIT;
+
+    static {
+        Calendar calendar = Calendar.getInstance();
+        calendar.add(DAY_OF_YEAR, -configurationService.getIntProperty("anonymise_statistics.time_limit", 90));
+        TIME_LIMIT = format(calendar, DATE_FORMAT_8601);
+    }
+
+    private AnonymizeStatistics() {
+
+    }
+
+
+    public static void main(String... args) throws ParseException {
+
+        parseCommandLineOptions(createCommandLineOptions(), args);
+        anonymiseStatistics();
+    }
+
+    private static Options createCommandLineOptions() {
+
+        Options options = new Options();
+        options.addOption(
+                builder(HELP_OPTION)
+                        .longOpt("help")
+                        .desc("Print the usage of the script")
+                        .hasArg(false)
+                        .build()
+        );
+
+        options.addOption(
+                builder(SLEEP_OPTION)
+                        .longOpt("sleep")
+                        .desc("Sleep a certain time between each solr request")
+                        .hasArg(true)
+                        .build()
+        );
+
+        options.addOption(
+            builder(BATCH_SIZE_OPTION)
+                .longOpt("batch")
+                .desc("The amount of Solr records to be processed per batch (defaults to 100)")
+                .hasArg(true)
+                .build()
+        );
+
+        options.addOption(
+            builder(THREADS_OPTION)
+                .longOpt("threads")
+                .desc("The amount of threads used by the script (defaults to 2")
+                .hasArg(true)
+                .build()
+        );
+
+        return options;
+    }
+
+    private static void parseCommandLineOptions(Options options, String... args) throws ParseException {
+
+        CommandLine commandLine = new DefaultParser().parse(options, args);
+
+        if (commandLine.hasOption(HELP_OPTION)) {
+            printHelp(options);
+            System.exit(0);
```

Comment:
Just a suggestion, not a change request: I would expect a negative exit code here. If the help parameter is given, all other parameters are ignored and the task stopped without having done anything. That doesn't seem to be a successful run to me.

Reply:
Good suggestion, I've changed it.